llvm/include/llvm/Transforms/Vectorize/LoopVectorizationLegality.h

```diff
@@ -538,6 +542,12 @@ class LoopVectorizationLegality {
   /// BFI and PSI are used to check for profile guided size optimizations.
   BlockFrequencyInfo *BFI;
   ProfileSummaryInfo *PSI;
+
+  /// If we discover function calls within the loop which have a valid
+  /// vectorized variant, record that fact so that LoopVectorize can
+  /// (potentially) make a better decision on the maximum VF and enable
+  /// the use of those function variants.
+  bool VecVariantsFound = false;
```

Comment:
Perhaps this is just me, but I find the name a bit too vague because it doesn't refer to variants of what exactly. How about calling it something like 'VecCallVariantsFound'?